src/SW_Domain.c

```diff
@@ -0,0 +1,153 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "include/SW_Domain.h"
+#include "include/filefuncs.h"
+#include "include/SW_Files.h"
+#include "include/SW_Times.h"
+
+/* =================================================== */
+/*             Local Function Definitions              */
+/* --------------------------------------------------- */
+
+/**
+ * @brief Read `domain.in` and report any problems encountered when doing so
+ *
+ * @param[in,out] SW_Model Struct of type SW_MODEL holding basic time
```

Comment:
SW_MODEL cannot be an input to `SW_DOM_read()` -- think of `SW_MODEL` describing one out of many possible tasks/runs that are contained in SW_DOMAIN